sycl/include/sycl/handler.hpp

```diff
@@ -524,18 +535,12 @@ class __SYCL_EXPORT handler {
   /// According to section 4.7.6.11. of the SYCL specification, a local accessor
   /// must not be used in a SYCL kernel function that is invoked via single_task
   /// or via the simple form of parallel_for that takes a range parameter.
-  template <typename KernelName, typename KernelType>
-  void throwOnLocalAccessorMisuse() const {
-    using NameT =
-        typename detail::get_kernel_name_t<KernelName, KernelType>::name;
-    using KI = sycl::detail::KernelInfo<NameT>;
-
-    auto *KernelArgs = &KI::getParamDesc(0);
-
-    for (unsigned I = 0; I < KI::getNumParams(); ++I) {
-      const detail::kernel_param_kind_t &Kind = KernelArgs[I].kind;
+  void throwOnLocalAccessorMisuse(
+      const std::vector<detail::kernel_param_desc_t> &ParamDescs) const {
```

Comment:
Why do we request kernel param descriptors from the caller, instead of just asking for the `KernelName` as before?

Reply:
To avoid fetching the parameter descriptors multiple times, the callers will need to obtain the descriptors anyway. Thanks to your comment, I noticed that `ParamDescs` weren't actually reused for `StoreLambda` in one of those cases. Updated.

Reply:
Why do we provide default parameter value for `StoreLambda`? Is this optimization really that important? And if it is, why not make it `constexpr`? Then we'd be able to just pass the type for readability/maintainability without runtime price...

Reply:
> Why do we provide default parameter value for StoreLambda?

So that we fetch the descriptors implicitly in the other cases. We need to check the descriptors before calling `StoreLambda` only in a couple of submit paths.
> Is this optimization really that important? And if it is, why not make it constexpr? Then we'd be able to just pass the type for readability/maintainability without runtime price...

Not sure if I follow. Are you suggesting to ditch the vector and use a constexpr constructible type to fetch the descriptors? I think the problem with that is the non-constexpr `KernelInfo::getParamDesc`.

Reply:
How wasn't all of that an issue before? Were we calling `getParamDesc` multiple times? And if so, why changing that as part of this PR?

Reply:
Agreed to drop this, and, if necessary, address it separately.